src/uwtools/drivers/forecast.py

```diff
@@ -231,14 +231,47 @@ def _define_boundary_files(self) -> Dict[str, str]:
 
         return boundary_files
 
+    def _prepare_and_run_batch_script(self, run_directory: str) -> bool:
+        """
+        Prepares and runs batch script.
+        """
+        batch_script = self.batch_script()
+        assert self._batch_script is not None
+        outpath = Path(run_directory) / self._batch_script
+
+        if self._dry_run:
+            # Apply switch to allow user to view the run command of config.
+            # This will not run the job.
+            log.info("Batch Script:")
+            batch_script.dump(None)
+            return True
+
+        batch_script.dump(outpath)
+        return self.scheduler.submit_job(outpath)
+
     def _prepare_config_files(self, run_directory: Path) -> None:
         """
         Collect all the configuration files needed for FV3.
         """
-
-        self.create_field_table(run_directory / "field_table")
-        self.create_model_configure(run_directory / "model_configure")
-        self.create_namelist(run_directory / "input.nml")
+        if self._dry_run:
+            for call in ("field_table", "model_configure", "input.nml"):
+                log.info(f"Would prepare: {run_directory}/{call}")
+        else:
+            self.create_field_table(run_directory / "field_table")
+            self.create_model_configure(run_directory / "model_configure")
+            self.create_namelist(run_directory / "input.nml")
+
+    def _prepare_and_execute_command(self) -> bool:
```

Comment:
This PR adds some new methods, so we should write targeted unit tests for those. Even if we have 100% coverage due to the fact that these methods get called by other methods that we _do_ have targeted test for, each method should really have at least one test that calls it directly. See e.g. the tests recently added for the `_RocotoXML` class: There's a test, for each method in the class, that explicitly, directly invokes that method, independent of anything else.